web-console/src/utils/sampler.ts

```diff
@@ -198,6 +201,8 @@ function makeSamplerIoConfig(
   } else if (specType === 'kinesis') {
     ioConfig = deepSet(ioConfig, 'useEarliestSequenceNumber', sampleStrategy === 'start');
   }
+  // In order to prevent potential data loss null columns should be kept by the sampler and shown in the ingestion flow
+  ioConfig = deepSet(ioConfig, 'inputFormat.keepNullColumns', true);
```

Comment:
Does this mean it's hard coded to true on the web console? Is there ever a case we want it to be false?

Reply:
Yes and no. It is hard coded to true on the sampleSpec but not the spec being created by the web console that is ingested into druid. This changes the web console so that all the columns including those that are null for all the values in the sample are shown to the user and keep during the ingestion flow, and will end up on the task spec that is generated by the web console. This prevents columns that only have null values from being excluded from being removed by the web console if it has data that just doesn't appear in the sample. So we would never want keepNullColumns be false on the sampleSpec because it could cause data to be lost. (you can actually see this happening in the changes I made to the e2e test for the web console)